core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -241,4 +249,34 @@ public String toString() {
     // Capture manifests which can be closed after scan planning
     return CloseableIterable.combine(splitTasks, ImmutableList.of(fileScanTasks));
   }
+
+  /**
+   * To be able to make refinements {@link #select(Collection)} and {@link #caseSensitive(boolean)} in any order,
+   * we resolve the schema to be projected lazily here.
+   *
+   * @return the Schema to project
+   */
+  private Schema lazyColumnProjection() {
```

Comment:
Why not put this implementation in `schema`?

Reply:
Will fix in separate PR.